src/envoy/tcp/alpn_proxy/alpn_proxy_test.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "common/buffer/buffer_impl.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/envoy/tcp/alpn_proxy/alpn_proxy.h"
+#include "src/envoy/tcp/alpn_proxy/alpn_proxy_initial_header.h"
+#include "src/envoy/tcp/alpn_proxy/header/alpn_proxy_header.pb.h"
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/network/mocks.h"
+
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Tcp {
+namespace AlpnProxy {
+namespace {
+bool protoMapEq(const ProtobufWkt::Struct& obj,
```

Comment:
MessageDifferencer is public now, you can find a use of it in envoy.

Reply:
Thanks..Fixed